### PR TITLE
Fix redshift and s3 for python3

### PR DESCRIPTION
**Pull request by gpoulin**
Also reactivate s3 unittest but skip broken tests

### CODE REVIEW

**Comment by gpoulin** [comment]
@Tarrasch @erikbern https://travis-ci.org/spotify/luigi/jobs/51993896#L754 ... Spotify doesn't want to upgrade to 2.7? :P I'm fixing this and Travis will be back to green :D

**Comment by Tarrasch** [comment]
@gpoulin, Spotify is such a big company that movement always have more friction. We've done lots of efforts going towards 2.7 though. :)

Anyway. Is the failing tests expected?

**Comment by Tarrasch** [comment]
i.e. Should I merge? :)

**Comment by gpoulin** [comment]
@Tarrasch It's not expected... it was supposed to bring back Travis to green, but the failure is with the doc and I didn't touch that... I just try to re-push to see.

File: test/s3_test.py [hunk 1]
```diff
@@ -83,16 +89,16 @@ def test_read(self):
         t = S3Target('s3://mybucket/tempfile', client=client)
         read_file = t.open()
         file_str = read_file.read()
-        self.assertEqual(self.tempFileContents, file_str)
+        self.assertEqual(self.tempFileContents, file_str.encode('utf8'))
```
Comment (berkerpeksag) [inline test/s3_test.py]: Nitpick: Actually, utf8 is not a valid identifier. Python has an "utf8" alias for "utf-8", but I think it's a mistake in Python's part. I wouldn't use "utf8" in new code.

**Comment by gpoulin** [comment]
The doc was building with sphinx 1.3 (https://travis-ci.org/spotify/luigi/jobs/51993893#L128) and now it tries to build with 1.2 (https://travis-ci.org/spotify/luigi/jobs/52010160)

**Comment by Tarrasch** [comment]
Thanks @gpoulin, Can you look at @berkerpeksag comment and then let me know when/if I should merge?

**Comment by gpoulin** [comment]
@Tarrasch I incorporated the "nitpick". If you don't see any issue you can merge.

**Comment by Tarrasch** [comment]
Thank you so much for all your help gpoulin! I fixed the merge conflict myself this time. :)

**Pull request merged**